### ISSUE TITLE
Exception thrown when order product contains configurable_fields (select_box_options)

### ISSUE THREAD

**Issue by teecord** [worstone/BigCommerce4Net#16]
I just got started with this code base and have been thankful for it. -Thank you :)
I found a workaround to this problem but any additional feedback would be greatly appreciated. 

When I execute the line (`var products = Client.OrdersProducts.Get(136);` I get an exception thrown only when the product contains configurable_fields. The exception is thrown from line 47 in NewtonSoftJsonDeserializer.cs (`return JsonConvert.DeserializeObject<T>(response.Content);`). 

An example value for `response.Content` which causes the error follows:
`[{"id":43,"order_id":136,"product_id":121,"order_address_id":37,"name":"MySuperC0oolProduct","sku":"","type":"physical","base_price":"25.0000","price_ex_tax":"25.0000","price_inc_tax":"27.0600","price_tax":"2.0600","base_total":"25.0000","total_ex_tax":"25.0000","total_inc_tax":"27.0600","total_tax":"2.0600","weight":"8.0000","quantity":1,"base_cost_price":"0.0000","cost_price_inc_tax":"0.0000","cost_price_ex_tax":"0.0000","cost_price_tax":"0.0000","is_refunded":false,"quantity_refunded":0,"refund_amount":"0.0000","return_id":0,"wrapping_name":"","base_wrapping_cost":"0.0000","wrapping_cost_ex_tax":"0.0000","wrapping_cost_inc_tax":"0.0000","wrapping_cost_tax":"0.0000","wrapping_message":"","quantity_shipped":0,"event_name":null,"event_date":"","fixed_shipping_cost":"0.0000","ebay_item_id":"","ebay_transaction_id":"","option_set_id":null,"parent_order_product_id":null,"is_bundled_product":false,"bin_picking_number":"","applied_discounts":[],"product_options":[],"configurable_fields":[{"id":4,"configurable_field_id":3,"name":"Product SerialNumber","fieldtype":"text","value":"MySerialNumber","original_filename":"","select_box_options":[""]}]}]`

The error details follow:
*{"Unexpected character encountered while parsing value: [. Path '[0].configurable_fields[0].select_box_options', line 1, position 1156."}*

To fix the issue, I removed the brackets from the `select_box_options` parameter before deserializing (`"select_box_options":""`)

**Comment by worstone** [comment]
I have not looked at BC json/xml data structure coming back in sometime but you need a Product object that has matching properties to deserialize to.  My guess is select_box_options is not defined in the object model/domain  or is not a collection, it needs to be a collection such as a list to deserialize correctly.

http://json2csharp.com/ will give you a clue on the Product object model/domain, just paste your json in and it spits out an C# object back.  This is what the JsonConvert.DeserializeObject<T> is expecting T = Product object in this case.

Your fix will will work fine, unless your need access to the select_box_options collection or the json coming back contains a collection the thing may blow up.

Hope that helps

**Comment by teecord** [comment]
Thanks for the speedy reply. I looked into it a bit further based on your feedback and believe I found the root cause and fix.

The *OrdersProductsConfigurableField* class does have a `select_box_options` property of type `string`. The logical assumptions are that 1) this should be a list, and 2) is only used when the configurable_field is of type `Select Box`. Since I'm using type `Short Text` my fix worked for me.

When I change the line `public virtual string SelectBoxOptions { get; set; } ` to `public virtual List<string> SelectBoxOptions { get; set; } ` in the *OrdersProductsConfigurableField* class I was able to run the original JSON without error. 

Strangely, I was not able to find mention of `select_box_options` in bigcommerce's documentation. 

Thanks again for your help!
